api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditQueryServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package org.pmiops.workbench.actionaudit;
+
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.stream.StreamSupport;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchConfig.ActionAuditConfig;
+import org.pmiops.workbench.model.AuditLogEntriesResponse;
+import org.pmiops.workbench.model.AuditLogEntry;
+import org.pmiops.workbench.utils.FieldValues;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ActionAuditQueryServiceImpl implements ActionAuditQueryService {
+
+  private BigQueryService bigQueryService;
+  private Provider<WorkbenchConfig> workbenchConfigProvider;
+  private static final String WORKSPACE_EVENTS_QUERY_STRING_FORMAT =
+      "SELECT\n"
+          + "  TIMESTAMP_MILLIS(CAST(jsonPayload.timestamp AS INT64)) as event_time,\n"
+          + "  jsonPayload.agent_type AS agent_type,\n"
+          + "  CAST(jsonPayload.agent_id AS INT64) AS agent_id,\n"
+          + "  jsonPayload.agent_email AS agent_username,\n"
+          + "  jsonPayload.action_id AS action_id,\n"
+          + "  jsonPayload.action_type AS action_type,\n"
+          + "  jsonPayload.target_type AS target_type,\n"
+          + "  CAST(jsonPayload.target_id AS INT64) AS target_id,\n"
+          + "  jsonPayload.target_property AS target_property,\n"
+          + "  jsonPayload.prev_value AS prev_value,\n"
+          + "  jsonPayload.new_value AS new_value\n"
+          + "FROM %s\n"
+          + "WHERE jsonPayload.target_id = %d AND\n"
```

Comment:
I would use a named parameter for any values injected from the client. We use the QueryParameterUtil  to help with this in CB land. Example: https://github.com/all-of-us/workbench/blob/c516872973fe331af315a0d4ef1f99da76347455/api/src/main/java/org/pmiops/workbench/cohortbuilder/SearchGroupItemQueryBuilder.java#L401-L401

Reply:
Brilliant! I was wondering what those were for. Also, now I can make an enough with my column names and friendly names and and setter methods on the model class like `AuditQueryParam.AGENT_ID.map(row, logentry)`. Similar to `WorkspaceTargetProperty`.